src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -262,20 +262,90 @@ private async Task<List<ContentSection>> TemplateFromRelease(Guid? releaseId)
                 });
         }
 
+        public async Task<Either<ActionResult, DeleteDataFilePlan>> GetDeleteDataFilePlan(Guid releaseId,
+            string subjectTitle)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanUpdateRelease)
+                .OnSuccess(async _ =>
+                {
+                    var subject = await _subjectService.GetAsync(releaseId, subjectTitle);
+                    var dependentDataBlocks = GetDependentDataBlocks(releaseId, subject.Id);
+                    
+                    return new DeleteDataFilePlan
+                    {
+                       DependentDataBlocks = dependentDataBlocks.
+                           Select(block => new DependentDataBlock
+                           {
+                               Name = block.Name,
+                               ContentSectionHeading = GetContentSectionHeading(block)
+                           })
+                           .ToList()
+                     };
+                });
+        }
+
+        private string GetContentSectionHeading(DataBlock block)
+        {
+            var section = block.ContentSection;
+
+            if (section == null)
+            {
+                return null;
+            }
+
+            switch (block.ContentSection.Type)
+            {
+                case ContentSectionType.Generic: return section.Heading;
+                case ContentSectionType.ReleaseSummary: return "Release Summary";
+                case ContentSectionType.Headlines: return "Headlines";
+                case ContentSectionType.KeyStatistics: return "Key Statistics";
+                case ContentSectionType.KeyStatisticsSecondary: return "Key Statistics";
+                default: return block.ContentSection.Type.ToString();
+            }
+        }
+
+
         public async Task<Either<ActionResult, IEnumerable<FileInfo>>> DeleteDataFilesAsync(Guid releaseId, string fileName, string subjectTitle)
         {
             return await _persistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanUpdateRelease)
                 .OnSuccess(async _ =>
                 {
+                    var subject = await _subjectService.GetAsync(releaseId, subjectTitle);
+
                     await _tableStorageService.DeleteEntityAsync("imports",
```

Comment:
I probably did it like this originally but I think the last thing that should be deleted is the imports table as this it what get's polled by the frontend so if the others fail then we'd still have the opportunity to delete things from the admin frontend.

Reply:
Yup happy to change this.  I'll put this into my 356 PR if that's OK?

Reply:
Rejigged order in https://github.com/dfe-analytical-services/explore-education-statistics/pull/1374